### PR TITLE
Add in GitlabCI Runner support

### PR DESCRIPTION
**Pull request by davidcollom**
This pull request (as titled) adds in runner support:

* List users runners
* List all runners (if admin)
* List projects runners (specific)
* Get runner
* Update runner
* Delete runner
* Enable project runner
* Disable project runner

### CODE REVIEW

File: lib/gitlab/client/runners.rb [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    #
+    # @example
+    #   Gitlab.project_disable_runner(2, 42)
+    #
+    # @param  [Integer, String] id The ID of a project.
+    # @param  [Integer, String] id The ID of a runner.
+    # @return <Gitlab::ObjectifiedHash>
+    def project_disable_runner(project_id, id)
+      body = { runner_id: id }
+      delete("/projects/#{project_id}/runners/#{id}", body: body)
```
Comment (asedge) [inline lib/gitlab/client/runners.rb]: Request body is not needed to disable a runner according to this doc: https://github.com/gitlabhq/gitlabhq/blob/master/doc/api/runners.md.  Have you observed something different?

File: lib/gitlab/client/runners.rb [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    #
+    # @param  [Hash] options A customizable set of options.
+    # @option options [String] :scope The scope of specific runners to show, one of: active, paused, online; showing all runners if none provided
+    # @return [Array<Gitlab::ObjectifiedHash>]
+    def runners(options = {})
+      if options[:scope]
+        get("/runners", query: options)
+      else
+        get("/runners")
+      end
```
Comment (asedge) [inline lib/gitlab/client/runners.rb]: Do we need the check here?  Couldn't we just pass the empty hash down regardless with no ill effect?

Comment (davidcollom) [inline lib/gitlab/client/runners.rb]: I was following the convention from https://github.com/NARKOZ/gitlab/blob/master/lib/gitlab/client/projects.rb#L15

```
    def projects(options={})
      if options[:scope]
        get("/projects/#{options[:scope]}", query: options)
      else
        get("/projects", query: options)
      end
    end
```
but I guess not.

File: lib/gitlab/client/runners.rb (hunk 1 above)
Comment (davidcollom) [inline lib/gitlab/client/runners.rb]: Agree'd I've left this in from the enable - will remove.

File: lib/gitlab/client/runners.rb (hunk 2 above)
Comment (asedge) [inline lib/gitlab/client/runners.rb]: Fair enough. :)

**Comment by asedge** [comment]
Thanks for the PR!  What is ` spec/fixtures/runners_edit.json` used for?  It's an empty file that does not appear to be referenced anywhere in the code.

**Comment by davidcollom** [comment]
`spec/fixtures/runners_edit.json` was added by mistake, I decided to use `spec/fixtures/runner.json` seen as though the response was technically no different.

File: lib/gitlab/client/runners.rb [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    #
+    # @param  [Hash] options A customizable set of options.
+    # @option options [String] :scope The scope of runners to show, one of: specific, shared, active, paused, online; showing all runners if none provided
+    # @return [Array<Gitlab::ObjectifiedHash>]
+    def all_runners(options = {})
+      if options[:scope]
+        get("/runners/all", query: options)
+      else
+        get("/runners/all")
+      end
```
Comment (asedge) [inline lib/gitlab/client/runners.rb]: Can you change this also so it's like the `runners` method?

**Comment by asedge** [comment]
@davidcollom One more thing to fix up and then you can go ahead and squash commits.

**Comment by asedge** [comment]
Thanks for the contribution!

**Pull request merged**